src/renderer/src/hooks/stores/mapeoDeviceStore.ts

```diff
@@ -0,0 +1,122 @@
+import { create } from 'zustand'
+
+type DeviceType = 'mobile' | 'desktop'
+
+export type Role = 'coordinator' | 'participant' | 'waitingInvite'
+
+type MapeoDevice =
+  | {
+      name: string
+      deviceId: string
+      deviceType: DeviceType
+      pastDeviceNames?: string[]
+      role: Extract<Role, 'waitingInvite'>
+    }
+  | {
+      name: string
+      deviceId: string
+      deviceType: DeviceType
+      pastDeviceNames?: string[]
+      role: Extract<Role, 'coordinator' | 'participant'>
+      dateAdded: Date
+    }
+
+type MapeoDeviceState = {
+  devices: Record<string, MapeoDevice>
```

Comment:
wondering if it would be helpful to separate member devices from non-member devices and updating the actions + hooks accordingly e.g.

```ts
members: Record<...>
nonMembers: Record<...>
```

having the separation and granularity could be helpful for certain actions (working with smaller data), but maybe not a huge benefit otherwise 🤷

Reply:
yeah not sure how useful it is in our case, but thinking that going from small to large is easy with Zustand via custom hooks and properly defined selectors. Can create granular hooks and compose them for more aggregated data if needed:

```ts
// Only care about non-members? Use this!
// Only changes when state.nonMembers updates
const useNonMembers() {
  return useStore(state => Object.values(state.nonMembers))
}

// Only care about members? Use this!
// Only changes when state.members updates
const useMembers() {
  return useStore(state => Object.values(state.members))
}

// Care about both members and non-members? Use this!
const useAllDevices() {
  const nonMembers = useNonMembers()
  const useMembers = useMembers()
  
  return [...nonMembers, ...member]
}

// Or maybe this is more Zustand-ish? idk
const useAllDevicesAlternative() {
  return useStore(state => [...Object.values(state.nonMembers), ...Object.values(state.members)])
}
```